### PR TITLE
Also lift Dot22Scalar.

### PR DESCRIPTION
**Pull request by abergeron**

### CODE REVIEW

File: theano/sandbox/gpuarray/opt.py [hunk 1]
```diff
@@ -751,6 +751,14 @@ def local_gpua_dot22(node, context_name):
 
 
 @register_opt('fast_compile')
+@op_lifter([tensor.blas.Dot22Scalar])
+def local_gpua_dot22scalar(node, context_name):
+    dot = gpu_dot22(as_gpuarray_variable(node.inputs[0], context_name),
+                    as_gpuarray_variable(node.inputs[1], context_name))
+    return [node.inputs[2] * dot]
```
Comment (nouiz) [inline theano/sandbox/gpuarray/opt.py]: why not convert this to a gemm call with alloc_empty?

Add a test to make sure it don't get broken.

Comment (abergeron) [inline theano/sandbox/gpuarray/opt.py]: Yes, I should (to both).

**Comment by nouiz** [comment]
travis is failing:

```
ERROR (theano.gof.opt): SeqOptimizer apply local_dot_to_dot22

ERROR (theano.gof.opt): Traceback:

ERROR (theano.gof.opt): Traceback (most recent call last):

  File "/home/travis/build/Theano/Theano/theano/gof/opt.py", line 230, in apply

    sub_prof = optimizer.optimize(fgraph)

  File "/home/travis/build/Theano/Theano/theano/gof/opt.py", line 89, in optimize

    ret = self.apply(fgraph, *args, **kwargs)

  File "/home/travis/build/Theano/Theano/theano/gof/opt.py", line 1886, in apply

    nb += self.process_node(fgraph, node)

  File "/home/travis/build/Theano/Theano/theano/gof/opt.py", line 1822, in process_node

    self.failure_callback(e, self, repl_pairs, lopt, node)

  File "/home/travis/build/Theano/Theano/theano/gof/opt.py", line 1680, in warn_inplace

    return NavigatorOptimizer.warn(exc, nav, repl_pairs, local_opt, node)

  File "/home/travis/build/Theano/Theano/theano/gof/opt.py", line 1666, in warn

    raise exc

Exception: Failed to infer_shape from Op Dot22.

Input shapes: [(TensorConstant{1}, Shape_i{0}.0), (Shape_i{0}.0, Shape_i{1}.0)]

Exception encountered during infer_shape: <type 'exceptions.NameError'>

Exception message: global name 'inputs_shapes' is not defined

Traceback: Traceback (most recent call last):

  File "/home/travis/build/Theano/Theano/theano/tensor/opt.py", line 930, in get_node_infer_shape

    [self.shape_of[r] for r in node.inputs])

  File "/home/travis/build/Theano/Theano/theano/tensor/blas.py", line 1755, in infer_shape

    return [[inputs_shapes[0][0], inputs_shapes[1][1]]]

NameError: global name 'inputs_shapes' is not defined
```

**Comment by abergeron** [comment]
Changes done.

**Comment by nouiz** [comment]
Did you run all new tests? If so, we can merge when travis pass.

**Comment by abergeron** [comment]
Yes, they pass locally.

**Pull request merged**